### ISSUE TITLE
Nicotine+ not working with latest MacOS Monteray 

### ISSUE THREAD

**Issue by recordstorejerk** [nicotine-plus/nicotine-plus#1660]
Just updated my Mac to the latest OS Monteray and Nicotine+ no longer works. The application boots up, there's a window outline where it should be, but isn't and the top action bar populates. Is it dead for the new OS?

<img width="1393" alt="Screen Shot 2021-11-03 at 11 21 39 AM" src="https://user-images.githubusercontent.com/93676446/140169818-0341f2e9-7a1e-4151-b454-86b2fdeb7830.png">
<img width="1265" alt="Screen Shot 2021-11-03 at 11 21 49 AM" src="https://user-images.githubusercontent.com/93676446/140169827-ba990228-fe79-410c-8eee-a78498727d6a.png">

**Comment by slook** [comment]
Thank you for the report @recordstorejerk please can specify what version of Nicotine+ you are running and method of install type?

**Comment by mathiascode** [comment]
Most likely this GTK bug: https://gitlab.gnome.org/GNOME/gtk/-/issues/4342
We have to wait for GTK 3.24.31.

What happens if you disable the header bar (Menu -> View -> Use Header Bar)?

**Comment by recordstorejerk** [comment]
Hey @mathiascode - looks like you're right - I have a grey bar at the top and a black screen. Gah! 

![Screen Shot 2021-11-03 at 1 16 01 PM](https://user-images.githubusercontent.com/93676446/140186072-3fbbc473-c684-47f4-8c4e-db77804faac0.png)

**Comment by francisuk1989** [comment]
@mathiascode do you think GTK4 would have the same issue?

**Comment by mathiascode** [comment]
> @mathiascode do you think GTK4 would have the same issue?

Not sure, but I don't see any bug reports for GTK4.

**Comment by francisuk1989** [comment]
In that case then @recordstorejerk if MacOS Monteray is homebrew compatible, you could try running Nicotine+ in Homebrew https://brew.sh

```
brew install nicotine-plus
NICOTINE_GTK_VERSION=4 nicotine
```

**Comment by slook** [comment]
What is the meaning of `"-s"` on Line 32 in the Homebrew Formulae?

https://github.com/Homebrew/homebrew-core/blob/c4afdfbc730ffc04cde7588b42adbbde105f3f00/Formula/nicotine-plus.rb#L32

`      exec bin/"nicotine", "-s"`

Is that because it is for testing or some other purpose for the syntax of exec? I don't get it.

  _-s, --hidden          start the program without showing window_

**Comment by slook** [comment]
> Most likely this GTK bug: https://gitlab.gnome.org/GNOME/gtk/-/issues/4342
> We have to wait for GTK 3.24.31.

Possible workaround:
https://gitlab.gnome.org/GNOME/gtk/-/issues/4395

**Comment by hboetes** [comment]
What happens if you start nicotine+ from a terminal? Can you paste the output that appear in the terminal?

**Comment by slook** [comment]
@hboetes A different Monteray user's console output looks normal, but that issue is slightly different because there is no gui at all with the brew install, just console log output as one would expect to see: https://github.com/nicotine-plus/nicotine-plus/discussions/1675#discussioncomment-1598376

**Comment by recordstorejerk** [comment]
@francisuk1989 - I tried what you suggested, but alas - same black screen. :(

**Comment by frozenpandaman** [comment]
I'm not even getting a black screen, just a crash after installing it via Homebrew:
```
$ nicotine                                                       
fish: Job 1, 'nicotine' terminated by signal SIGSEGV (Address boundary error)
```

Tried @francisuk1989's instructions above and getting a way uglier error:
```
$ NICOTINE_GTK_VERSION=4 ./nicotine
[2021-11-16 00:33:09] Loading GTK 4.4.1
/path/to/nicotine-plus/pynicotine/gtkgui/frame.py:204: Warning: g_object_weak_unref: couldn't find weak ref 0x1054be4a8(0x104e85a60)
  self.application.add_window(self.MainWindow)
/path/to/nicotine-plus/pynicotine/gtkgui/frame.py:204: Warning: g_object_weak_unref: couldn't find weak ref 0x1054be4a8(0x104e85a40)
  self.application.add_window(self.MainWindow)
/path/to/nicotine-plus/pynicotine/gtkgui/frame.py:204: Warning: g_object_weak_unref: couldn't find weak ref 0x1054be4a8(0x104e85300)
  self.application.add_window(self.MainWindow)
/path/to/nicotine-plus/pynicotine/gtkgui/frame.py:204: Warning: g_object_weak_unref: couldn't find weak ref 0x1054be4a8(0x104e85360)
  self.application.add_window(self.MainWindow)
/path/to/nicotine-plus/pynicotine/gtkgui/frame.py:204: Warning: g_object_weak_unref: couldn't find weak ref 0x1054be4a8(0x104f344c0)
  self.application.add_window(self.MainWindow)
/path/to/nicotine-plus/pynicotine/gtkgui/frame.py:204: Warning: g_object_weak_unref: couldn't find weak ref 0x1054be4a8(0x104f05c60)
  self.application.add_window(self.MainWindow)
/path/to/nicotine-plus/pynicotine/gtkgui/frame.py:204: Warning: g_object_weak_unref: couldn't find weak ref 0x1054be4a8(0x104e85260)
  self.application.add_window(self.MainWindow)
/path/to/nicotine-plus/pynicotine/gtkgui/frame.py:204: Warning: g_object_weak_unref: couldn't find weak ref 0x1054be4a8(0x104e85220)
  self.application.add_window(self.MainWindow)
/path/to/nicotine-plus/pynicotine/gtkgui/frame.py:204: Warning: g_object_weak_unref: couldn't find weak ref 0x1054be4a8(0x104f35440)
  self.application.add_window(self.MainWindow)
/path/to/nicotine-plus/pynicotine/gtkgui/frame.py:204: Warning: g_object_weak_unref: couldn't find weak ref 0x1054be4a8(0x104f35960)
  self.application.add_window(self.MainWindow)
/path/to/nicotine-plus/pynicotine/gtkgui/frame.py:204: Warning: g_object_weak_unref: couldn't find weak ref 0x1054be4a8(0x104e85340)
  self.application.add_window(self.MainWindow)
/path/to/nicotine-plus/pynicotine/gtkgui/frame.py:204: Warning: g_object_weak_unref: couldn't find weak ref 0x1054be4a8(0x104e852e0)
  self.application.add_window(self.MainWindow)
/path/to/nicotine-plus/pynicotine/gtkgui/frame.py:204: Warning: g_object_weak_unref: couldn't find weak ref 0x1054be4a8(0x104f33c20)
  self.application.add_window(self.MainWindow)
/path/to/nicotine-plus/pynicotine/gtkgui/frame.py:204: Warning: g_object_weak_unref: couldn't find weak ref 0x1054be4a8(0x104e85a20)
  self.application.add_window(self.MainWindow)
/path/to/nicotine-plus/pynicotine/gtkgui/frame.py:204: Warning: g_object_weak_unref: couldn't find weak ref 0x1054be4a8(0x104f35d60)
  self.application.add_window(self.MainWindow)
[2021-11-16 00:33:10] Loading Nicotine+ 3.2.0.dev1, using Python 3.9.8 (main, Nov 10 2021, 03:48:35) 
[Clang 13.0.0 (clang-1300.0.29.3)]
[2021-11-16 00:33:10] Listening on port: 2234
[2021-11-16 00:33:10] Loading plugin system
[2021-11-16 00:33:10] You need to specify a username and password before connecting…
[2021-11-16 00:33:10] UPnP: Failed to forward external port 2234: [Errno 35] Resource temporarily unavailable
Fatal Python error: Segmentation fault

Thread 0x0000000171507000 (most recent call first):
  File "/opt/homebrew/Cellar/python@3.9/3.9.8/Frameworks/Python.framework/Versions/3.9/lib/python3.9/threading.py", line 316 in wait
  File "/opt/homebrew/Cellar/python@3.9/3.9.8/Frameworks/Python.framework/Versions/3.9/lib/python3.9/threading.py", line 574 in wait
  File "/opt/homebrew/Cellar/python@3.9/3.9.8/Frameworks/Python.framework/Versions/3.9/lib/python3.9/threading.py", line 1297 in run
  File "/opt/homebrew/Cellar/python@3.9/3.9.8/Frameworks/Python.framework/Versions/3.9/lib/python3.9/threading.py", line 973 in _bootstrap_inner
  File "/opt/homebrew/Cellar/python@3.9/3.9.8/Frameworks/Python.framework/Versions/3.9/lib/python3.9/threading.py", line 930 in _bootstrap

Thread 0x000000016f4ef000 (most recent call first):
  File "/path/to/nicotine-plus/pynicotine/slskproto.py", line 1476 in run
  File "/opt/homebrew/Cellar/python@3.9/3.9.8/Frameworks/Python.framework/Versions/3.9/lib/python3.9/threading.py", line 973 in _bootstrap_inner
  File "/opt/homebrew/Cellar/python@3.9/3.9.8/Frameworks/Python.framework/Versions/3.9/lib/python3.9/threading.py", line 930 in _bootstrap

Current thread 0x000000010316c580 (most recent call first):
  File "/opt/homebrew/lib/python3.9/site-packages/gi/overrides/Gio.py", line 42 in run
  File "/path/to/nicotine-plus/pynicotine/gtkgui/__init__.py", line 69 in run_gui
  File "/path/to/nicotine-plus/pynicotine/__init__.py", line 245 in run
  File "/path/to/nicotine-plus/./nicotine", line 24 in <module>
fish: Job 1, 'NICOTINE_GTK_VERSION=4 ./nicoti…' terminated by signal SIGSEGV (Address boundary error)
```

**Comment by slook** [comment]
> getting a way uglier error

Your log output looks like it could be very useful @frozenpandaman 

> I am using an M1/Apple Silicon device so that may be the cause of the issue.

It seems Gtk4 doesn't support Apple M1 yet, no one upstream predicted that special measures were going to be needed, because community developer documentation is sparse for this proprietary platform.

Not sure if this will be helpful or not... please can you repeat another log output with:
`NICOTINE_GTK_VERSION=4 G_ENABLE_DIAGNOSTIC=1 ./nicotine`

**Comment by mathiascode** [comment]
The crash is probably the same one as mentioned here, and affects both GTK 3 and GTK 4: https://github.com/nicotine-plus/nicotine-plus/issues/1475#issuecomment-944769368

It's only possible to run Nicotine+ via Rosetta 2 at the moment.

**Comment by mathiascode** [comment]
The unstable Nicotine+ builds should work fine now.

**Comment by recordstorejerk** [comment]
Which unstable Nicotine+ builds?

**Comment by mathiascode** [comment]
https://github.com/nicotine-plus/nicotine-plus/blob/master/doc/TESTING.md#macos-catalina1015-and-newer

**Comment by frozenpandaman** [comment]
> please can you repeat another log output with:
> `NICOTINE_GTK_VERSION=4 G_ENABLE_DIAGNOSTIC=1 ./nicotine`

@slook Maybe fixed now in the build below, but this gives me `UserWarning: resource_tracker: There appear to be 3 leaked semaphore objects to clean up at shutdown`. [Here](https://www.toptal.com/developers/hastebin/helowegobi)'s the full log.


> https://github.com/nicotine-plus/nicotine-plus/blob/master/doc/TESTING.md#macos-catalina1015-and-newer

@mathiascode Running the .app non-natively is fine, as long as it runs at all! Will try out that build, thank you!

**Comment by mathiascode** [comment]
@S-n-d Are you on macOS Monterey? What is the output if you run Nicotine+ in the terminal (right click the Nicotine+ launcher -> Show Package Contents -> navigate to "Contents/MacOS" -> start the "Nicotine+" executable)?

**Comment by S-n-d** [comment]
@mathiascode correct. terminal output:

`/Applications/Nicotine+.app/Contents/MacOS/Nicotine+ ; exit;
objc[51427]: Class GNotificationCenterDelegate is implemented in both /Applications/Nicotine+.app/Contents/MacOS/libgio-2.0.0.dylib (0x10caa7330) and /usr/local/Cellar/glib/2.68.4/lib/libgio-2.0.0.dylib (0x10d8f72f0). One of the two will be used. Which one is undefined.

(org.nicotine_plus.Nicotine:51427): GLib-GIO-CRITICAL **: 15:02:56.797: g_application_run: assertion 'G_IS_APPLICATION (application)' failed

Saving session...completed.

[Process completed]
`

**Comment by mathiascode** [comment]
Seems to be this issue: https://github.com/pyinstaller/pyinstaller/issues/6354

TLDR: Nicotine+ crashes on Monterey if a Homebrew GTK installation is present on the system. There seems to be a proposed fix for gobject-introspection. https://gitlab.gnome.org/GNOME/gobject-introspection/-/merge_requests/303

**Comment by mathiascode** [comment]
@S-n-d Could you check if the latest unstable build is working now?

**Comment by mathiascode** [comment]
Closing since the patch was merged: https://github.com/Homebrew/homebrew-core/pull/90116

**Comment by S-n-d** [comment]
Can confirm it's working now! Thanks!

**Comment by dosch** [comment]
Just downloaded Nicotine+ (version 3.1.1). 
I have the same issue as OP.

M1 Chip Mac with MacOS 12.1 (Monterey)

**Comment by neofright** [comment]
I can also confirm that version 3.2.0.rc2 available from [here](https://nicotine-plus.org/doc/TESTING.html#macos-catalina1015-and-newer) works on an M1 Mac on macOS 12.1.

Many thanks to all contributors of this project.
